src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/CompilerToVM.java

```diff
@@ -476,6 +476,24 @@ Object[] resolveBootstrapMethod(HotSpotConstantPool constantPool, int cpi) {
 
     private native Object[] resolveBootstrapMethod(HotSpotConstantPool constantPool, long constantPoolPointer, int cpi);
 
+    /**
+     * Gets the constant pool index of a static argument of a {@code CONSTANT_Dynamic_info} or
+     * @{code CONSTANT_InvokeDynamic_info} entry. Used when the list of static arguments in the
+     * {@link BootstrapMethodInvocation} is a {@code List<PrimitiveConstant>} of the form
+     * {{@code arg_count}, {@code pool_index}}, meaning the arguments are not already resolved and that
+     * the JDK has to lookup the arguments when they are needed. The {@code cpi} corresponds to
+     * {@code pool_index} and the {@code index} has to be smaller than {@code arg_count}.
+     *
+     * @param cpi the index of a {@code CONSTANT_Dynamic_info} or @{code CONSTANT_InvokeDynamic_info} entry
+     * @param index the index of the static argument in the list of static arguments
```

Comment:
What happens when `index` is out of bounds?

Reply:
Currently, it returns another constant pool index or a random number. I can add a check in the `CachedBSMArgs.get` method to prevent this.

Reply:
After double checking, there is in fact already an `assert` for checking if the index is out of bounds. Should I change it to an exception?

Reply:
Is the assertion in code that can throw exceptions?

Reply:
The assertion is [here](https://github.com/openjdk/jdk/blob/bd477810b176696e0fd043f5594663ebcf9884cf/src/hotspot/share/oops/constantPool.hpp#L621-L625). I don't think we can throw exception there. I could check the size in `jvmciCompilerToVM.bootstrapArgumentIndexAt`, but it would require a bit of code duplication to obtain it.

Reply:
Ok. You should just add a note in the javadoc that the result is undefined if `index` is out of bounds. This is all internal API so I assume we can guarantee the value is correct?

Reply:
Yes, it is only used in `CachedBSMArgs.get`, which will throw an `ArrayIndexOutOfBoundsException` before we arrive there.